x-pack/plugins/maps/common/descriptor_types/layer_descriptor_types.ts

```diff
@@ -28,7 +33,11 @@ export type LayerDescriptor = {
   areLabelsOnTop?: boolean;
   minZoom?: number;
   maxZoom?: number;
-  sourceDescriptor: AbstractSourceDescriptor | null;
+  sourceDescriptor:
```

Comment:
The maps plugin exposes a `registerSource` method, allowing plugins to register custom sources. Prior to custom sources, `sourceDescriptor` used to be the union of all source descriptor types. Now, the typing is specifically only contains the minimum required.

Where is this causing typing concerns? Can you cast  your sourceDescriptor to add more typing specificity?

Reply:
reverted this change